docs/Architecture.md

```diff
@@ -1,42 +1,55 @@
-└──────┘ └───┘          │ └─────────────────┘    └─────────┘
-                        │
-                        │ ┌──────────────┐ ┌───────────────┐
-                        │ │Management API│ │Runtime Watcher│
-                        │ └──────────────┘ └───────────────┘
-```
+#### Management API
 
--   Management API (Public API): This is the API that manages the scheduler and rooms. It is the component the users (usually game engineers) use while interacting with Maestro, making updates, and fetching scheduler information;
--   Rooms API (Internal API): API used by the game rooms to report their status to Maestro and communicate any extra information that may apply;
--   Operations executor worker: Worker that takes queued operations and processes them;
--   Runtime watcher: Watches any state change that happens on the Runtime and takes the necessary action (like updating some Maestro state or triggering an operation);
+Management API is the module responsible for receiving user requests. It accepts gRPC and HTTP requests. It could be used to fetch a **Scheduler** status or else to execute a change in them. There is some different kind of routes. In the **Scheduler** creation, it simply adds a new entry in the Postgres (and the worker posteriorly creates a respective queue to it). In the fetch **Scheduler** routes, it fetches from Postgres the **Scheduler** information and in the Redis the **Operations**/game rooms information. And in the change routes, it creates **Operations** and enqueues them in the respective **Scheduler** queue in the Redis. Finally, fetch **Operation**/game rooms routes it fetches its information from Redis.
 
-## Code structure
-Maestro codebase is organized following some concepts of Hexagonal architecture (Ports & Adapters). This "style" was used as a base of the development and had some adaptations. For example, the API is not considered a primary/input Port. Instead, it is outside the core package.
+![Management API IMAGE](./diagrams/Architecture-Management-API.jpg)
 
-Most of the business logic is present at services, and they rely on the ports to not depend on the specific implementations.
+#### Rooms API
 
-Entities are the structs that hold data on the system, and they don't provide any logic besides its initialization and validation. Therefore, services and ports should "communicate" using entities so that the entire system "speaks" the same language regarding data structs.
+Rooms API is the module that receive requests from game rooms and executes according to which kind of request was done.
 
-The API is responsible for converting the external protocol (currently gRPC with grpc-gateway) to entities, invoking the necessary service, and converting the response back.
+> ⚠ Note: The requests that Maestro receives in the Rooms API are documented in [this proto file](https://github.com/topfreegames/protos/blob/master/maestro/grpc/protobuf/events.proto).
```

Comment:
Actually, I think this is not right... These proto are used to events forwarded by Maestro. The Rooms-API requests are the ones under OpenAPI - RoomsService. Also, I think it's relevant to mention the Maestro-Client as a useful tool for interacting with it